nwg_panel/modules/executor.py

```diff
@@ -81,55 +82,46 @@ def __init__(self, settings, icons_path, executor_name):
             Gdk.threads_add_timeout_seconds(GLib.PRIORITY_LOW, settings["interval"], self.refresh)
 
     def update_widget(self, output):
```

Comment:
The length of `output` can be either 1 or 2:
1. contains either icon label or icon path
2. a tuple of `(icon_path, label)`

The aims here are to fill in corresponding widgets and hide the non-existent one when `len(output) == 1`.
The logic was a mess and looks much better now.